### ISSUE TITLE
snakemake workflow for roads and rail does not work

### ISSUE THREAD

**Issue by itrcrisks** [nismod/open-gira#190]
Got the error:

Building DAG of jobs...
MissingRuleException:
No rule to produce results/kenya-latest_filter-road/edges.geoparquet (if you use input functions make sure that they don't raise unexpected exceptions).

**Comment by tomalrussell** [comment]
Just testing this command, which has processed okay:

```
snakemake -c16 results/kenya-latest_filter-road-primary/edges.gpq
```

Two big differences:
- `.gpq` not `.geoparquet` for the file extension - this is defined in the snakemake rule file ` workflow/transport/join_network.smk`
- `filter-road-primary` not `filter-road` - the filters are defined in files in `config/osm_filters` and contain [filter expressions](https://docs.osmcode.org/osmium/latest/osmium-tags-filter.html#filter-expressions) for `osmium tags-filter`

**Comment by tomalrussell** [comment]
To review - should be closed by #191

**Comment by albert8v** [comment]
Hi,

I tried snakemake -c16 results/spain-latest_filter-rail/nodes.gpq
Got an error

Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 16
Rules claiming more threads will be scaled down.
Select jobs to execute...
[Mon Nov  4 14:50:48 2024]
Error in rule download_osm:
    jobid: 0
    output: results/input/OSM/spain-latest.osm.pbf

RuleException:
KeyError in line 10 of /lustre/soge1/users/cenv0972/AFP_python/open-gira/workflow/transport/openstreetmap.smk:
'spain-latest'
  File "/lustre/soge1/users/cenv0972/AFP_python/open-gira/workflow/transport/openstreetmap.smk", line 10, in __rule_download_osm
  File "/lustre/soge1/users/cenv0972/micromamba/envs/open-gira/lib/python3.10/concurrent/futures/thread.py", line 58, in run
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-11-04T145042.447841.snakemake.log

**Comment by tomalrussell** [comment]
I think you found this already, but just to record it here:

The informative part of the error is:

```
KeyError in line 10 of /lustre/soge1/users/cenv0972/AFP_python/open-gira/workflow/transport/openstreetmap.smk:
'spain-latest'
```

which is failing to find `spain-latest` in the list of OSM extracts in [`config/config.yaml`](https://github.com/nismod/open-gira/blob/9ec11cbecd321c8a048929c08ce95663f2184cbc/config/config.yaml#L6-L38)

The simplest fix it to add a line for spain: 

```python
    spain-latest: 'https://download.geofabrik.de/europe/spain-latest.osm.pbf'
```

You can browse https://download.geofabrik.de/ to see all the available geographical extracts.

**Comment by tomalrussell** [comment]
Closed by #191
